src/main/scala/analysis/Analysis.scala

```diff
@@ -22,7 +22,7 @@ trait Analysis[+R]:
 
 /** Base class for value analysis with simple (non-lifted) lattice.
   */
-trait ConstantPropagation(val program: Program) {
+trait ConstantPropagation(val program: Program, val assumeR31: Boolean) {
```

Comment:
What is the purpose of assumeR31? It doesn't seem to actually be used anywhere, and seems like it would just confuse anything involving the stack if it were to be used?